### ISSUE TITLE
systemd unit dbus subscriptions should use unit events.

### ISSUE THREAD

**Issue by james-lawrence** [coreos/go-systemd#229]
basically this TODO needs to be implemented.
```func (s *SubscriptionSet) Subscribe() (<-chan map[string]*UnitStatus, <-chan error) {
	// TODO: Make fully evented by using systemd 209 with properties changed values
	return s.conn.SubscribeUnitsCustom(time.Second, 0,
		mismatchUnitStatus,
		func(unit string) bool { return s.filter(unit) },
	)
}```

can't accurately monitor units being restarted without calling SubscribeUnitsCustom with a very short delay. which wastes a bunch of cpu.

**Comment by purpleidea** [comment]
My https://github.com/purpleidea/mgmt/ project uses this fine library. One beautiful aspect of it is that when there's work to do, it wakes up and very quickly does that work. When it's quiet, everything should be sleeping and not consuming CPU. I noticed it's consuming quite a lot of CPU when it's meant to be idle.

I captured some data. Here's the flamegraph.

<img width="2040" height="858" alt="Image" src="https://github.com/user-attachments/assets/e273ab04-d025-477d-8e2b-c0b60f97d244" />

TL;DR: virtually 100% of the CPU is caused by this polling. This is quite a significant amount on my machines, the process is typically floating between 3% and 15% in `top`.

That loop is here: https://github.com/coreos/go-systemd/blob/a5cf58b14cbbe07725f2e387d0ca07c96764fc76/dbus/subscription.go#L126

This really should switch to being properly event based. Hopefully this data helps motivate one of the big companies using this to contribute a fix.

**Comment by purpleidea** [comment]
Note the related comment here: https://github.com/coreos/go-systemd/blob/a5cf58b14cbbe07725f2e387d0ca07c96764fc76/dbus/subscription_set.go#L36
